### PR TITLE
DTSPO-6396 Rename + add uami 

### PR DESCRIPTION
**Pull request by zc-hmcts**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by endakelly — APPROVED** [review]
Need to update this line as well but all good otherwise https://github.com/hmcts/azure-policy/blob/3d82d11f53106071c012b20e960cc980ebe08cb3/assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json#L16

File: assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json [hunk 1]
```diff
@@ -28,6 +28,6 @@
         "tier": "Free"
     },
     "type": "Microsoft.Authorization/policyAssignments",
-    "id": "/providers/Microsoft.Management/managementGroups/dts002/providers/Microsoft.Authorization/policyAssignments/HMCTSAppGWDiagnostic",
-    "name": "HMCTSAppGWDiagnostic"
+    "id": "/providers/Microsoft.Management/managementGroups/dts002/providers/Microsoft.Authorization/policyAssignments/HMCTSAGWDiagnosticGlobal",
+    "name": "HMCTSAGWDiagnosticGlobal"
```
Comment (timja) [inline assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:32]: is it renamed because of a length limitation or as a better name?

Comment (zc-hmcts) [reply assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:32]: just bringing it in line with the way the others are named

File: assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json [hunk 2]
```diff
@@ -13,21 +13,18 @@
...
+        "policyDefinitionId": "/providers/Microsoft.Management/managementGroups/dts002/providers/Microsoft.Authorization/policyDefinitions/HMCTSAGWDiagnostic",
         "scope": "/providers/Microsoft.Management/managementGroups/dts002"
     },
     "location": "uksouth",
     "identity": {
-        "type": "UserAssigned",
-        "userAssignedIdentities": {
-            "/subscriptions/8ae5b3b6-0b12-4888-b894-4cec33c92292/resourceGroups/soc-core-infra-prod-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/soc-prod-eventhub-azure-policy": {}
-        }
+        "type": "SystemAssigned"
```
Comment (timja) [inline assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:21]: why switch?

Comment (zc-hmcts) [reply assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:21]: need the Azure Event Hubs Data Owner” role, raised a pr to add the role to the uami but there’s a script to check there’s no owner roles ..and we want least privilege for that so seemed better to just use system assigned here

Comment (timja) [reply assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:21]: can probably just remove that check it's very outdated and was more for the exact role 'Owner' not other ones

Comment (zc-hmcts) [reply assignments/mgmt-groups/mg-dts002/assign.appgateway_diagnostics.json:21]: the uami was originally made bc the system assigned only grants permissions to the target subscription which doesn't work if the event hub is in another sub. that issue doesn't apply here so are we wanting to use a uami just for consistency?

**Pull request merged**